CMakeLists.txt

```diff
@@ -0,0 +1,662 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+# Ubuntu 22.04 LTS Jammy Jellyfish, https://wiki.ubuntu.com/Releases, EOSS in June 2027:
+#  - CMake 3.22.1, https://packages.ubuntu.com/jammy/cmake
+#
+# Centos Stream 9, EOL in May 2027:
+#  - CMake 3.26.5, https://mirror.stream.centos.org/9-stream/AppStream/x86_64/os/Packages/
+cmake_minimum_required(VERSION 3.22)
+if(POLICY CMP0141)
+  # MSVC debug information format flags are selected by an abstraction.
+  # We want to use the CMAKE_MSVC_DEBUG_INFORMATION_FORMAT variable
+  # to select the MSVC debug information format.
+  cmake_policy(SET CMP0141 NEW)
+endif()
+
+if (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+  message(FATAL_ERROR "In-source builds are not allowed.")
+endif()
+
+#=============================
+# Project / Package metadata
+#=============================
+set(PACKAGE_NAME "Bitcoin Core")
+set(CLIENT_VERSION_MAJOR 27)
```

Comment:
In a2317e27b7fb86df4e32cd1674c06e09cb808248 "cmake: Add root `CMakeLists.txt` file"

The plan is to have this merged post 28.x branch off, so I think this should be set to 28.

Reply:
Correct. Will be updated in a follow up PR, along with a few other amendments such as https://github.com/bitcoin/bitcoin/pull/30712.